### ISSUE TITLE
Avisynth scripts .avs playing is broken again

### ISSUE THREAD

**Issue by Divetoxx** [mpv-player/mpv#1730]
Last windows build doesn't play .avs files.

[lavf] avformat_open_input() failed
Failed to recognize file format.

**Comment by lachs0r** [comment]
I blame FFmpeg. There were some completely unchecked AviSynth-related changes recently.
Does the current build (just updated) fix this? If not, can you try with [AviSynth+](http://avs-plus.net/)?

**Comment by Divetoxx** [comment]
No, It doesn't. AviSynth+ too :(

**Comment by qyot27** [comment]
Which version?  AviSynth+ r1825 works fine for me, both on 32-bit Windows and 64-bit Wine.  FFmpeg and mpv builds from the 26th, so those changes are also present.

**Comment by Divetoxx** [comment]
I'm sorry, I tested r1576. Where can I get binary r1825 for Windows?

**Comment by qyot27** [comment]
I'm using one of my own builds, but ultim did post a build of r1779:
http://forum.doom9.org/showthread.php?p=1713428#post1713428

While there are *some* changes to the actual code between 1779 and 1825, most of them are actually the conversion of the English documentation from HTML to RST/Sphinx.

**Comment by qyot27** [comment]
I think I know what's happening, though.  The changes to the C interface API in 2.6 RC1 (which AviSynth+ merged ca. r1718) mean that builds prior to that - the 2.6 alphas included - no longer work with libavformat, no matter if it's FFmpeg or Libav.  I tested 2.6 RC1 and it's fine, but Plus r1576 was released in December 2013 (read: it still uses the same C interface API as 2.6a5, which should also fail).

And probably totally unrelated to this is that the 2.5 escape hatch in Libav stopped working at some point prior to FFmpeg merging the same bit of code in, which is why the error it's supposed to throw ("AviSynth 2.5.8 is no longer supported, please upgrade to 2.6.") when there's an attempt to use 2.5 never shows up.  It doesn't show the error message in avconv either.

**Comment by Divetoxx** [comment]
Avisynth 2.6.0 RC 1 [150114] works fine!

**Comment by lachs0r** [comment]
I’ve added a note to the [list of changes for the 2015-03-24 build](https://mpv.srsfckn.biz/changes/2015-03-24/), for future reference.

@qyot27: Are there any plans on restoring compatibility with old AviSynth versions? Otherwise I’d just close this, since newer versions should still be perfectly compatible with old plugins and scripts.

**Comment by qyot27** [comment]
Restoring compatibility in FFmpeg?  I doubt that it'll be restored at this point (but I'm not really in control of that; I jumped in because the patch I submitted on the Libav side got merged into FFmpeg prematurely, and I went into damage control mode).  nevcairiel reverted the changes in the modified branch that LAV Filters uses, so that commit could be cherry-picked on an *ad hoc* basis before compilation if a user still wants to build with 2.5 support.

None of the options look particularly good, IMO.  The demuxer as it is right now will mostly straighten itself out once AviSynth+ 0.2 is released, although I have no idea what the timeframe is on that (some of the talk in the forum thread and the rstification pull request lead me to think it may be a 'soon').  Since with a proper 0.2, users won't accidentally download a version that's too old, and the classic AviSynth POV for 2.6 has been that as soon as the next alpha or RC is out, the older ones shouldn't be used anymore (so the alphas that sit between 2.5.8 and 2.6 RC1 can and should be ignored when considering this).

Which just leaves 2.5 support itself, and that's untenable to continue supporting alongside the use of newer headers, as it would become a maintenance burden on the FFMS2 compat header and also become a potential licensing minefield.  So it would require reverting all the AviSynth-related commits to FFmpeg from the past week, but that would simply re-introduce the more egregious disparities between FFmpeg's and Libav's AviSynth demuxer (Libav not shipping local copies was a minor issue compared to the thought of Libav using AVSC_API for some functions and FFmpeg going *back* to using AVSC_INLINE versions for those same functions).  *If* it gets reverted, the most responsible thing to do is keep a modified version checker around that merely warns 2.5 users that 2.5 support is outdated and will eventually be dropped, but doesn't error out on them.  But that still needs whatever's causing the version checker to not display the upgrade message to be fixed (which needs to be done anyway, of course).

tl;dr: yeah, I'd say go ahead and close it.

**Comment by wm4** [comment]
>tl;dr: yeah, I'd say go ahead and close it.

No new developments, so, ok.

**Comment by qyot27** [comment]
The error message that prompts users to upgrade works again with FFmpeg HEAD.
